dm/syncer/syncer.go

```diff
@@ -1321,6 +1321,20 @@ func (s *Syncer) syncDDL(queueBucket string, db *dbconn.DBConn, ddlJobChan chan
 		if !ok {
 			return
 		}
+
+		// set timezone
+		if ddlJob.timezone != "" {
+			setTimezoneSQL := fmt.Sprintf("SET SESSION TIME_ZONE = '%s'", ddlJob.timezone)
+			ddlJob.ddls = append([]string{setTimezoneSQL}, ddlJob.ddls...)
+			setTimezoneSQLDefault := fmt.Sprintf("SET SESSION TIME_ZONE = DEFAULT")
```

Comment:
maybe we should not revert it to default. For two sequential DDL, if the latter one uses the same timezone as the former one, binlog will not write timezone again. So for latter one we should keep the timezone

Please check mysqlbinlog's output to see if it's right.

Reply:
Yes. If the time zone is the same as that of last time, the time zone information will not be found in binlog.